### PR TITLE
FOGL-1562 Config map based readings  + refactoring 

### PR DESCRIPTION
**Pull request by praveen-garg**

### CODE REVIEW

**Comment by praveen-garg** [comment]
The listening TCP port 502 is reserved for MODBUS , However, some markets or applications might require that another port is dedicated to MODBUS over TCP.

File: python/foglamp/plugins/south/modbustcp/modbustcp.py [hunk 1]
```diff
@@ -26,49 +32,84 @@
...
-        'default':     '127.0.0.1'
+        'type': 'string',
+        'default': '127.0.0.1'
+    },
+    'sourcePort': {
+        'description': 'The port of the Modbus TCP data source',
+        'type': 'integer',
+        'default': '502'
+    },
+    'entitiesMap': {
```
Comment (MarkRiddoch) [inline python/foglamp/plugins/south/modbustcp/modbustcp.py]: This does not match the requested name in the Jira ticket and makes it different from the other modbus plugin we have. What not follow the request?

File: python/foglamp/plugins/south/modbustcp/modbustcp.py [hunk 2]
```diff
@@ -26,49 +32,84 @@
...
+        'description': 'Modbus entities map',
+        'type': 'JSON',
+        'default': json.dumps({
+            "coils": {},
+            "discreteInputs": {},
```
Comment (MarkRiddoch) [inline python/foglamp/plugins/south/modbustcp/modbustcp.py]: Again this differs from the requested name.

File: python/foglamp/plugins/south/modbustcp/modbustcp.py [hunk 3]
```diff
@@ -26,49 +32,84 @@
...
+            "holdingRegisters": {
```
Comment (MarkRiddoch) [inline python/foglamp/plugins/south/modbustcp/modbustcp.py]: This is also different from the requested name.

**Review by MarkRiddoch — CHANGES_REQUESTED** [review]

File: python/foglamp/plugins/south/modbustcp/modbustcp.py (hunk 1 above)
Comment (praveen-garg) [reply python/foglamp/plugins/south/modbustcp/modbustcp.py]: we can stick to 
```
"map" : {  "description" : "Modbus register map", 
                        "type" : "JSON", "default" : { 
                                "coils" : { }, 
                                "inputs" : { }, 
                                "registers" : { },
                                "inputRegisters" : { }
                        }
```
if its a MUST need. entitiesMap was kept as map is a inbuilt function name in python.

Also for inputs-> discreteInputs, registers > holdingResgisters are moe standard. Hence I kept that way. I would have pointed that out once we have final PR. 

anyways, do you think we should stick to map and inputs, registers as key?!

**Comment by praveen-garg** [comment]
@MarkRiddoch  Please let me know if we are good with these naming for entities map or we want to keep as in JIRA. 

I have tested with dummy values using:

```
ModSim64.zip (816K)
x64 Version for Windows 7
(July 3, 2013)
```

 https://www.win-tech.com/html/modsim32.htm

[screenshot for holding registers]
tested with dummy values

**Comment by praveen-garg** [comment]
[deploy-modbus-tcp-plugin.sh.zip](https://github.com/foglamp/foglamp-south-modbustcp/files/2220841/deploy-modbus-tcp-plugin.sh.zip)

**Comment by MarkRiddoch** [comment]
The Jira was written to make this consistent with the C based plugin that does Modbus RTU and Modbus TCP, hence the choice of the names given in the Jira. That consistency is important, so if there is a good reason to change we need to do it in both places.

**Comment by praveen-garg** [comment]
Now its consistent with the C based plugin. Please review.

**Review by MarkRiddoch — APPROVED** [review]

File: make_deb [hunk 4]
```diff
@@ -109,8 +109,8 @@ mkdir "${package_name}"
 echo -n "Populating the package and updating version file..."
 cd "${package_name}"
 cp -R ${GIT_ROOT}/packages/Debian/${architecture}/* .
-sed -i "s/Version: 0.0/Version: $version/g" DEBIAN/control
-sed -i "s/Depends: foglamp/Depends: foglamp (=$foglamp_version)/g" DEBIAN/control
+sed -i "s/Version: 0.0.0/Version: $version/g" DEBIAN/control
```
Comment (ashish-jabble) [inline make_deb]: ${version}

File: make_deb [hunk 5]
```diff
@@ -109,8 +109,8 @@ mkdir "${package_name}"
...
+sed -i "s/Depends: foglamp/Depends: foglamp ($foglamp_version)/g" DEBIAN/control
```
Comment (ashish-jabble) [inline make_deb]: ${foglamp_version}

**Review by ashish-jabble — CHANGES_REQUESTED** [review]

**Review by ashish-jabble — APPROVED** [review]

**Pull request merged**